compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1117,6 +1119,12 @@ fn extra_compiler_flags() -> Option<(Vec<String>, bool)> {
     if !result.is_empty() { Some((result, excluded_cargo_defaults)) } else { None }
 }
 
+// Set parallel mode before thread pool creation as the session will already create locks.
+#[allow(rustc::bad_opt_access)]
+fn set_parallel_mode(sopts: &config::Options) {
```

Comment:
Maybe put this in `rustc_interface` and reuse it for `rustdoc`?

Reply:
Done :)